packages/vuetify/src/composables/filter.ts

```diff
@@ -135,7 +135,12 @@ export function useFilter <T extends { value: unknown }> (
 ) {
   const filteredItems: Ref<T[]> = ref([])
   const filteredMatches: Ref<Map<unknown, Record<string, FilterMatch>>> = ref(new Map())
-  const transformedItems = computed(() => options?.transform ? unref(items).map(options?.transform) : unref(items))
+  const transformedItems = computed(() => options?.transform
+    ? unref(items).map(item => ({
+      item,
+      ...(options?.transform?.(item) ?? {}),
```

Comment:
Not a huge fan of this. When implementing a custom filter function it would be `item.item.x` (without a spread). There is also a potential key conflict if table has a column called `item`.

Possible alternative solution.

```js
items.map(item => [item, options.transform(item)])
```

And then in filter function on L71

```
const [item, transformed] = items[i]
```

That way `transformed` could be used for `getPropertyFromItem` and `item` could be passed as is to `keyFilter/filter`

Reply:
Changes made. I'm still figuring out how I want to approach the tests.